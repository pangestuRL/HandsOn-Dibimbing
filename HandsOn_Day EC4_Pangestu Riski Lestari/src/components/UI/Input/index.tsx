import React from 'react';

const Input = (props: any) => {
  return (
    <>
      {props.type === 'textarea' ? (
        <textarea {...props} />
      ) : (
        <input {...props} />
      )}
    </>
  );
};

export default Input;
