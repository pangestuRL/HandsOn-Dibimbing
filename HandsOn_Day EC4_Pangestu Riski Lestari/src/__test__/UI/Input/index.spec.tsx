import Input from '@/components/UI/Input';
import { fireEvent, render } from '@testing-library/react';

describe('Input component test', () => {
  it('should render', () => {
    const { container } = render(<Input />);

    expect(container).toMatchSnapshot();
  });

  it('should render textarea', () => {
    const { container } = render(<Input type="textarea" />);

    expect(container).toMatchSnapshot();
  });

  it('should handle change', () => {
    const onChange = jest.fn();
    const { getByPlaceholderText } = render(
      <Input onChange={onChange} type="text" placeholder="username" />
    );
    const input = getByPlaceholderText('username');
    fireEvent.change(input, { target: { value: 'kyra@email.com' } });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(input).toHaveValue('kyra@email.com');
  });
});
