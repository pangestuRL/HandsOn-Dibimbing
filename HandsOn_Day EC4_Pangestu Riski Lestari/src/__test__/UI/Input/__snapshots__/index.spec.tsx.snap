// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component test should render 1`] = `
<div>
  <input />
</div>
`;

exports[`Input component test should render textarea 1`] = `
<div>
  <textarea
    type="textarea"
  />
</div>
`;
